### DISCUSSION TITLE
Windows does not exist.

### DISCUSSION THREAD

**Discussion by edwin1212256756** [winapps-org/winapps#349] [Support]
Installing is no problem, but when i open any app i get a notification in the right corner from winapps, saying: Windows does not exist.

**Comment by AkechiShiro** [comment]
Could you share more info on your setup ? Linux distro ? Windows is running in docker or VM ?

**Comment by edwin1212256756** [comment]
Well i use arch running libvirt. Windows 11. i use wayland but i tried changing to x11 but came to no help.

**Comment by AkechiShiro** [comment]
Do you have the config file (`~/.config/winapps/winapps.conf`) ?

Could you paste it's content here ?

**Comment by bonzoskill** [comment]
Hi @AkechiShiro , i have the same issue as i showed here: 
https://github.com/winapps-org/winapps/issues/301#issuecomment-2438526272

Anyway i past here some logs and info :

I'm using libvirt as backend, Arch as distribution, Wayland as DE and ZSH as Shell.
I followed the guide for the VM with libvirt and everything worked well.
I've git cloned the winapps repo and executed ./setup.sh for the install wizard. If using bash <(curl https://raw.githubusercontent.com/winapps-org/winapps/main/setup.sh) , it gives me this error :

% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
100 79583 100 79583 0 0 296k 0 --:--:-- --:--:-- --:--:-- 295k
################################################################################
WinApps Install Wizard

################################################################################

/proc/self/fd/13: line 208: ./install/inquirer.sh: No such file or directory
Exiting with status '1'.

Anyway if i git clone and run ./setup.sh i can install the apps i want and them appears on Arch launcher. When i try to lunch them this error occurs. (I tried to install for the User and System either but same error)

if i run set -x ./winapps > output.txt 2>&1 the output is zsh: permission denied: ./winapps
if i run set -x ./winapps/setup.sh > output.txt 2>&1 the output is

�[1m################################################################################
WinApps Install Wizard

################################################################################
�[0m
./winapps/setup.sh: line 208: ./install/inquirer.sh: No such file or directory
�[1;41;37mExiting with status '1'.�[0m




In the file ~/.local/share/winapps/winapps.log, the log is :

[Fri Oct 18 01:11:11 AM CEST 2024-13497] START
[Fri Oct 18 01:11:11 AM CEST 2024-13497] SCRIPT_DIR: /home/bonzo/.local/bin
[Fri Oct 18 01:11:11 AM CEST 2024-13497] SCRIPT_ARGS: word-o365
[Fri Oct 18 01:11:11 AM CEST 2024-13497] HOME_DIR: /home/bonzo
[Fri Oct 18 01:11:11 AM CEST 2024-13497] LAST_RUN: 1729008905
[Fri Oct 18 01:11:11 AM CEST 2024-13497] THIS_RUN: 1729206671
[Fri Oct 18 01:11:11 AM CEST 2024-13497] Using FreeRDP command 'xfreerdp3'.
[Fri Oct 18 01:11:11 AM CEST 2024-13497] ERROR: WINDOWS NONEXISTENT. EXITING.


the content of my config file `~/.config/winapps/winapps.conf` is this (i have obscured user and password) :

\##################################
\#   WINAPPS CONFIGURATION FILE   #
\##################################

\# INSTRUCTIONS
\# - Leading and trailing whitespace are ignored.
\# - Empty lines are ignored.
\# - Lines starting with '#' are ignored.
\# - All characters following a '#' are ignored.
\# [WINDOWS USERNAME]
RDP_USER="******"

\# [WINDOWS PASSWORD]
RDP_PASS="*******"

\# [WINDOWS DOMAIN]
\# DEFAULT VALUE: '' (BLANK)
RDP_DOMAIN=""

\# [WINDOWS IPV4 ADDRESS]
\# NOTES:
\# - If using 'libvirt', 'RDP_IP' will be determined by WinApps at runtime if left unspecified.
\# DEFAULT VALUE:
\# - 'docker': '127.0.0.1'
\# - 'podman': '127.0.0.1'
\# - 'libvirt': '' (BLANK)
RDP_IP=""

\# [WINAPPS BACKEND]
\# DEFAULT VALUE: 'docker'
\# VALID VALUES:
\# - 'docker'
\# - 'podman'
\# - 'libvirt'
\# - 'manual'
WAFLAVOR="libvirt"

\# [DISPLAY SCALING FACTOR]
\# NOTES:
\# - If an unsupported value is specified, a warning will be displayed.
\# - If an unsupported value is specified, WinApps will use the closest supported value.
\# DEFAULT VALUE: '100'
\# VALID VALUES:
\# - '100'

\# [ADDITIONAL FREERDP FLAGS & ARGUMENTS]
\# DEFAULT VALUE: '/cert:tofu /sound /microphone'
\# VALID VALUES: See https://github.com/awakecoding/FreeRDP-Manuals/blob/master/User/FreeRDP-User-Manual.markdown
RDP_FLAGS="/cert:tofu /sound /microphone"

\# [MULTIPLE MONITORS]
\# NOTES:
\# - If enabled, a FreeRDP bug *might* produce a black screen.
\# DEFAULT VALUE: 'false'
\# VALID VALUES:
\# - 'true'
\# - 'false'
MULTIMON="false"

\# [DEBUG WINAPPS]
\# NOTES:
\# - Creates and appends to ~/.local/share/winapps/winapps.log when running WinApps.
\# DEFAULT VALUE: 'true'
\# VALID VALUES:
\# - 'true'
\# - 'false'
DEBUG="true"

\# [AUTOMATICALLY PAUSE WINDOWS]
\# NOTES:
\# - This is currently INCOMPATIBLE with 'docker' and 'manual'.
\# - See https://github.com/dockur/windows/issues/674
\# DEFAULT VALUE: 'off'
\# VALID VALUES:
\# - 'on'
\# - 'off'
AUTOPAUSE="off"

\# [AUTOMATICALLY PAUSE WINDOWS TIMEOUT]
\# NOTES:
\# - This setting determines the duration of inactivity to tolerate before Windows is automatically paused.
\# - This setting is ignored if 'AUTOPAUSE' is set to 'off'.
\# - The value must be specified in seconds (to the nearest 10 seconds e.g., '30', '40', '50', etc.).
\# - For RemoteApp RDP sessions, there is a mandatory 20-second delay, so the minimum value that can be specified here is '20'.
\# - Source: https://techcommunity.microsoft.com/t5/security-compliance-and-identity/terminal-services-remoteapp-8482-session-termination-lo>
\# DEFAULT VALUE: '300'
\# VALID VALUES: >=20
AUTOPAUSE_TIME="300"

\# [FREERDP COMMAND]
\# NOTES:
\# - WinApps will attempt to automatically detect the correct command to use for your system.
\# DEFAULT VALUE: '' (BLANK)
\# VALID VALUES: The command required to run FreeRDPv3 on your system (e.g., 'xfreerdp', 'xfreerdp3', etc.).
FREERDP_COMMAND=""

**Comment by AK4EVA1** [comment]
Hi, I'm having the same issue, fresh install of Windows, followed the instructions correctly but it fails with error code 10 stating that RDPWindows (name of VM) does not exist..any help would be greatly appreciated and please let me know what you need from my end to assist in resolving this issue. thanks

**Comment by Zalaax** [comment]
> Hi, I'm having the same issue, fresh install of Windows, followed the instructions correctly but it fails with error code 10 stating that RDPWindows (name of VM) does not exist..any help would be greatly appreciated and please let me know what you need from my end to assist in resolving this issue. thanks

Also experiencing this issue while using libvirt under Fedora 41. I've also noticed that my VM is not listed when I run `virsh list --all` but will appear when I run the same command as sudo.

**Comment by brilliantnz** [comment]
@Zalaax do you set LIBVIRT_DEFAULT_URI="qemu:///system" to bashrc or zshrc ? vm installed in QEMU system or QEMU user session?

**Comment by Zalaax** [comment]
@imoize I know that the vm is installed as a user session, not sure about bashrc or zshrc as I'm not sure how to check

**Comment by Mizuna737** [comment]
Same issue here, on Arch using libvirt. Install works perfectly and clearly runs apps as the windows command prompt opens when it's scanning for apps.

**Comment by brilliantnz** [comment]
When installing VM make sure `export LIBVIRT_DEFAULT_URI="qemu:///system"` is set in .zshrc or .bashrc, so VM is installed in system session not in user. The next problem if when launching the application, a notification appears saying: Windows does not exist. That's because `LIBVIRT_DEFAULT_URI="qemu:///system"` env is not read as in the script `~/.local/bin/winapps` use shebang `#!/usr/bin/env bash`.

There are some workaround:
1. Try put `export LIBVIRT_DEFAULT_URI="qemu:///system"` in .bashrc

2. Try put `export LIBVIRT_DEFAULT_URI="qemu:///system"` directly in `~/.local/bin/winapps` script after copyright text

3. Put `LIBVIRT_DEFAULT_URI=qemu:///system` in global env /etc/environment

Try one by one which one works for you.

**Comment by bonzoskill** [comment]
> When installing VM make sure `export LIBVIRT_DEFAULT_URI="qemu:///system"` is set in .zshrc or .bashrc, so VM is installed in system session not in user. The next problem if when launching the application, a notification appears saying: Windows does not exist. That's because `LIBVIRT_DEFAULT_URI="qemu:///system"` env is not read as in the script `~/.local/bin/winapps` use shebang `#!/usr/bin/env bash`.
> 
> There are some workaround:
> 
>     1. Try put `export LIBVIRT_DEFAULT_URI="qemu:///system"` in .bashrc
> 
>     2. Try put `export LIBVIRT_DEFAULT_URI="qemu:///system"` directly in `~/.local/bin/winapps` script after copyright text
> 
>     3. Put `LIBVIRT_DEFAULT_URI=qemu:///system` in global env /etc/environment
> 
> 
> Try one by one which one works for you.

For me exporting directly in `~/.local/bin/winapps` solved the problem, i had the export in .zshrc but didn't work. Thanks

**Comment by Sithuk** [comment]
Thank you @bonzoskill That resolved the issue for me too. The .bashrc and /etc/environment locations didn't work.

**Comment by bonzoskill** [comment]
@Sithuk i have clean the installation and after reinstall i have no more the file in the location `~/.local/bin/winapps`. I added the line of code in `/etc/environment` and for me work too, the only one doesn't work is the first option in the file .zshrc

**Comment by Sithuk** [comment]
I’m not sure I logged out and back in again to trigger a reload of /etc/environment
So it might have worked there. I’ll need to re-test that.

**Comment by Laikulo** [comment]
I ran into this, and have an alternate workaround that doesn't affect other things using libvirt.

I placed
`export LIBVIRT_DEFAULT_URI='qemu:///system'` in `~/.config/winapps/winapps.conf`. This is technically leveraging unsafe parsing of said config, but it works.
